server/player_service.py

```diff
@@ -64,6 +64,18 @@ def clear_dirty(self):
 
     async def fetch_player_data(self, player):
         async with self._db.acquire() as conn:
+            result = await conn.execute(
+                select([
+                    user_group.c.technical_name,
+                    user_group.c.id
+                ])
+                .select_from(user_group_assignment.join(user_group))
+                .where(user_group_assignment.c.user_id == player.id)
+            )
+            player.user_groups = {
+                entry async for row in result for entry in row.as_tuple()
```

Comment:
I don't get that line, why we have to call for each entry as `row.as_tuple()?` when result row is always named tuple?
and how is it possible, to have loop in loop. Like, when i read the result from sqlalchemy, i will get result row as named tuple and it's it, no need to iterate. And when i wanna have the values, i'll just write something like:
```
player.user_groups {(row.technical_name, row.id) for row in result}
```
Also I don't get, how do you wanna work with that set? Do you flatten whole result to `player.user_groups`? Like there will be the `{technical_name, id, technical_name2, id2}`? And is there any reason why? Because only place where you're using it is in `if row.group_id in player.user_groups`, which could be solved on database side, like just query database is there is result with proper tables join.

Reply:
To be continued in 5-6 hours

Reply:
I've found peace of code that is working with `player.user_groups` below. And I'm thinking that you can leave that `player.user_groups` variable.

Reply:
Well it's not just a named tuple. It's a `RowProxy` object from sqlalchemy which does a bunch of stuff. I'm pretty sure just iterating it did not work so I had to call `as_tulpe()`.

> Do you flatten whole result to player.user_groups

yes exactly. I'm grabbing ids and technical_names so that I can use either to check permissions. This lets me write the permission query using only 1 join instead of 2.

Reply:
Ok.

Reply:
Btw I reverted this so it uses 2 joins but doesn’t pull as much data from the db each time.

Reply:
Oh maybe this part was in a different commit. I’ll need to check again